src/Agent.Worker/Build/GitSourceProvider.cs

```diff
@@ -280,14 +288,21 @@ public class GitSourceProvider : SourceProvider, ISourceProvider
             executionContext.Output($"Cleaning embeded credential from repository: {endpoint.Name} (Git)");
 
             Uri repositoryUrl = endpoint.Url;
-            string targetPath = executionContext.Variables.Get(Constants.Variables.Build.SourcesDirectory);
+            string targetPath;
```

Comment:
The variables class had tracing built-in when retrieving a variable. Can you add tracing to HostContext.Info the variable that was retrieved from the endpoint data?

Same comment applies to every source provider where you changed this.